### PR TITLE
distsql: embed ProcessorBase in planNodeToRowSource

### PR DESCRIPTION
**Pull request by asubiotto**
planNodeToRowSource had incorrect closing semantics. When closing itself
    (in any case), it would call `ConsumerClosed` on `toDrain` and proceed to
    drain it. The contract is that `Next` should never be called after
    `ConsumerClosed`. The underlying issue here is that we are trying to
    re-implement RowSource semantics. To solve this, and any other possible
    issues, we embed ProcessorBase and reuse existing code.

This change also adds an `EXPLAIN ANALYZE` test that was previously failing because of these closing semantics.

### CODE REVIEW

File: pkg/sql/distsqlrun/processors.go [hunk 1]
```diff
@@ -691,8 +691,10 @@ func (pb *ProcessorBase) moveToTrailingMeta() {
 	}
 
 	pb.State = stateTrailingMeta
-	if trace := getTraceData(pb.Ctx); trace != nil {
-		pb.trailingMeta = append(pb.trailingMeta, ProducerMetadata{TraceData: trace})
+	if pb.span != nil {
```
Comment (jordanlewis) [inline pkg/sql/distsqlrun/processors.go]: Was this broken before?

**Review by jordanlewis — APPROVED** [review]
This is fantastic, such good cleanup.

File: pkg/sql/distsqlrun/processors.go (hunk 1 above)
Comment (asubiotto) [reply pkg/sql/distsqlrun/processors.go]: No, we just always called `StartInternal` which would create a span for us. If we don't do this check, we `getTraceData` from the span in `pb.Ctx` which is the parent span so we will send along a `RecordedSpan` that will also be created down the line, resulting in a `duplicate span` error.

Comment (jordanlewis) [reply pkg/sql/distsqlrun/processors.go]: Ah, got it, cool

**Comment by asubiotto** [comment]
Fixed the lint issues, bors r+

**Pull request merged**